src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += num;
+
+    return adder;
```

Comment:
Consider adding a check to ensure that the input `num` is a number before adding it to `sum`. This will prevent potential issues if non-numeric values are passed to the `adder` function.